packages/lockfile-lint/src/config.js

```diff
@@ -0,0 +1,30 @@
+const debug = require('debug')('lockfile-lint')
+const {cosmiconfigSync} = require('cosmiconfig')
+const cli = require('../src/cli')
+
+const validateConfig = input =>
+  ['path', 'type', 'validateHttps', 'emptyHostname', 'allowedHosts', 'allowedSchemes'].reduce(
+    (config, key) => {
+      // eslint-disable-next-line security/detect-object-injection
+      if (input[key] !== undefined) {
+        // eslint-disable-next-line security/detect-object-injection
+        config[key] = input[key]
+      }
+
+      return config
+    },
+    {}
+  )
+
+const cosmiconfigResult = cosmiconfigSync('lockfile-lint').search()
+
+let fileConfig
+if (cosmiconfigResult) {
+  fileConfig = validateConfig(cosmiconfigResult.config)
```

Comment:
@mtlewis is this a synchronous operation? what happens if it throws?

Reply:
Yes, it's synchronous since we're using `cosmiconfigSync`. I hadn't spotted that it could throw though, thanks for pointing it out. Newest version of the code now handles errors, ignoring file-based config and logging an error.